### PR TITLE
feat: added video button

### LINKED ISSUE #1609: Adding a video button in SessionDetail Activity
**Actual Behaviour**

Currently the app doesnt make use of video url to enable watching live event.

**Expected Behaviour**

Adding video button

**Steps to reproduce it**

-

**LogCat for the issue**

-

**Screenshots of the issue**

-

**Would you like to work on the issue?**
Yes

**Target Milestone to solve the issue**

https://github.com/fossasia/open-event-android/milestone/27

### PR DESCRIPTION
**Pull request by arp95**
Fixes #1609 
Changes: 
Added a video button in the SessionDetailActivity.java.

Screenshots for the change: 
<details>
<summary>Click to view image</summary>

</details>

### CODE REVIEW

**Comment by arp95** [comment]
@iamareebjamal @mahikaw @shreyasnbhat @Shailesh351
please review.

File: android/app/src/main/res/layout/content_session_detail.xml [hunk 1]
```diff
@@ -159,6 +159,18 @@
...
                         android:textSize="@dimen/text_size" />
+
+                    <ImageButton
+                        android:id="@+id/watch"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_below="@+id/location"
+                        app:srcCompat="@drawable/ic_play"
+                        android:layout_centerHorizontal="true"
+                        android:paddingTop="28dp"
```
Comment (shreyasnbhat) [inline android/app/src/main/res/layout/content_session_detail.xml]: @arp95  Add these dimensions to `dimen.xml`

File: android/app/src/main/res/layout/content_session_detail.xml [hunk 2]
```diff
@@ -159,6 +159,18 @@
...
+                        android:paddingBottom="28dp"
```
Comment (shreyasnbhat) [inline android/app/src/main/res/layout/content_session_detail.xml]: Same here.

**Comment by Shailesh351** [comment]
What happens when user clicks on video button ?

**Comment by arp95** [comment]
yeah that is what i wanted to discuss.
as @iamareebjamal said the links might not be youtube links always. should i start an implicit intent then?

**Comment by iamareebjamal** [comment]
Yes @arp95 
Furthermore, if there is a youtube link, it was decided to show the thumnail on a card, which isn't present here?

**Comment by arp95** [comment]
@iamareebjamal
do you mean changing the background of the button if its a youtube link?

**Comment by iamareebjamal** [comment]
photo6185781353795135421

I'm talking about a thumbnail like this

**Comment by arp95** [comment]
@iamareebjamal do you mean replacing the image button with youtube thumbnail if a youtube video link?

File: android/app/src/main/res/layout/content_session_detail.xml (hunk 2 above)
Comment (arp95) [inline android/app/src/main/res/layout/content_session_detail.xml]: sure doing it.

**Comment by iamareebjamal** [comment]
Just instead of this, use a CardView instead, with image as background if it is youtube link or leave the card grey if it is not

**Comment by arp95** [comment]
@iamareebjamal got it. meaning the button will remain the same. only background change?
but wont the black cause visibility problems?

**Comment by iamareebjamal** [comment]
Make the button translucent grey. It will be visible on any background then, specially the default light grey card background

**Comment by arp95** [comment]
sure.

**Comment by Shailesh351** [comment]
I think it will look good and clean if you shift video button in appbar layout with gravity center.

**Comment by arp95** [comment]
Even i think the same. Let others confirm.

**Comment by iamareebjamal** [comment]
@Shailesh351 Most sessions don't have video link, so just hiding and showing card will be easy. But if you can handle the layout dynamically @arp95, it'd be great

**Comment by arp95** [comment]
@iamareebjamal can you please extend the milestones of both issues, that is this one and adding the twon pane layout one if possible?

**Comment by arp95** [comment]
wait i think it will be done by tomorrow.

**Comment by arp95** [comment]
guys please see the layout and also the click logic.

19072920_10211522290807373_1876712685_o

19072966_10211522290327361_614132131_o

Also @iamareebjamal it doesnt look good to include a thumbnail as the layout didnt look good. i think this way it looks better. What say?

**Comment by iamareebjamal** [comment]
The extra space when there is no video isn't looking good. Can you also add the screenshot with thumbnail?

**Comment by arp95** [comment]
@iamareebjamal sending it. Also taking a look to reduce the space dynamically.

**Comment by arp95** [comment]
19048725_10211529118258055_1454896769_o

point of showing this like this is. the background can be dark or normal so the thumbnail might not look good always. before one is fine then?

**Comment by Shailesh351** [comment]
@arp95 @iamareebjamal IMO showing  thumbnails here don't look good.

**Comment by iamareebjamal** [comment]
@arp95 Why not loading full res thumbnail in whole of the collapsing toolbar?

**Comment by arp95** [comment]
@iamareebjamal dont you think the visibilty of icons will become a problems as thumbnails might be dark.
Also the session title is over there, it might get affected as well.

**Comment by iamareebjamal** [comment]
@arp95 Please take a look

**Comment by arp95** [comment]
@iamareebjamal i know this is acheievable but then i need to change the layout of session detail where i have to bring the session title below the app bar layout. but harshit strictly told me to not change the layout of the file.

**Comment by iamareebjamal** [comment]
That's why I suggested to use a card instead, including rich imagery in views is encouraged and makes a great user experience

**Comment by arp95** [comment]
what should be a better choice then?

**Comment by iamareebjamal** [comment]
Please try a CardView showing the thumbnail

**Comment by arp95** [comment]
below the location right?

**Comment by iamareebjamal** [comment]
Yes

**Comment by arp95** [comment]
sure. showing you the result.

**Comment by arp95** [comment]
19104803_10211529496147502_227584029_o

@iamareebjamal hows this one looking?

**Comment by iamareebjamal** [comment]
Just make the play button translucent grey, so it appears over all kind of background, otherwise looks fine

Also, give an example of where youtube field is not present so as to show what will it look like without thumbnail

**Comment by arp95** [comment]
@iamareebjamal if i made it translucent grey the icon was not visible? 
yeah showing that example.

**Comment by arp95** [comment]
19022657_10211529891237379_287920926_o

19073167_10211529894317456_40981633_o

19073026_10211529891717391_1151164757_o

@iamareebjamal now see.

**Comment by iamareebjamal** [comment]
It's too translucent, that's why. Try some more alpha, see for yourself what is looking good

**Comment by iamareebjamal** [comment]
Also, decrease the height of collapsing toolbar layout back to normal please

**Comment by arp95** [comment]
yeah decreasing the height.
actually i found the black one better. anyways trying that

**Comment by arp95** [comment]
19105043_10211530582174652_304836968_o 1

guys please review this. @iamareebjamal @Shailesh351 @shreyasnbhat

**Comment by iamareebjamal** [comment]
This is solid grey @arp95 :/

**Comment by arp95** [comment]
@iamareebjamal i am using hexadecimal values from here. http://htmlcolorcodes.com/
can you tell which one are you looking for?

**Comment by iamareebjamal** [comment]
Are you adding alpha to it?

Try #99DFDFDF

**Comment by arp95** [comment]
18948713_10211530771219378_227047557_o

looks a tad light to me. should i make a bit darker?

**Comment by iamareebjamal** [comment]
Yeah sure

**Comment by arp95** [comment]
19073320_10211530909902845_799924532_o

now fine?

**Comment by iamareebjamal** [comment]
Yeah :+1:

**Comment by arp95** [comment]
Without video link:

19072991_10211531283232178_260808293_o

With video link but not youtube url:

19048685_10211531283312180_1269138956_o

With youtube link:

19095840_10211531283592187_1453344961_o

**Comment by arp95** [comment]
@iamareebjamal @Shailesh351 @shreyasnbhat @mahikaw 
please review the pr.

**Pull request merged**